### PR TITLE
chore: Standardize on the use of "status" vs "quality"

### PR DESCRIPTION
**Pull request by kingthorin**
Related to https://github.com/zaproxy/zaproxy/issues/4732

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
`ActiveScanAPI.createPluginProgressEntry` already has both `quality` and `status`. I believe the existing "status" could be renamed "state" but that would be a breaking change, so I wasn't sure if I should bother updating it or not. Other places I added a "status" element alongside the existing "quality" element.

File: zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 // ZAP: 2014/07/07 Issue 389: Enable technology scope for scanners
 // ZAP: 2014/10/25 Issue 1062: Made plugins that calls sendandrecieve also invoke scanner
 // hook before and after message update
-// ZAP: 2014/11/19 Issue 1412: Init scan rule status (quality) from add-on
+// ZAP: 2014/11/19 Issue 1412: Init scan rule status (status) from add-on
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java:47]: Does not seem worth changing, or remove one of them (same in other places).

File: zap/src/main/java/org/zaproxy/zap/extension/ascan/CategoryTableModel.java [hunk 2]
```diff
@@ -257,19 +257,30 @@ private List<PluginWrapper> getTestList() {
...
         public Plugin getPlugin() {
             return plugin;
         }
 
+        /**
+         * Returns the status.
+         *
+         * @return the status to be displayed.
+         * @since 2.12
+         */
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/ascan/CategoryTableModel.java:276]: Not necessary.

File: zap/src/main/java/org/zaproxy/zap/extension/ascan/CategoryTableModel.java [hunk 3]
```diff
@@ -257,19 +257,30 @@ private List<PluginWrapper> getTestList() {
...
+        public StatusUI getStatus() {
+            return status;
+        }
+
+        @Deprecated
         public StatusUI getQuality() {
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/ascan/CategoryTableModel.java:282]: Can be removed, it's a private class. Same for `ScannerWrapper`.

File: zap/zap.gradle.kts [hunk 4]
```diff
@@ -267,6 +267,7 @@ val japicmp by tasks.registering(JapicmpTask::class) {
         "org.zaproxy.zap.extension.api.PythonAPIGenerator#generatePythonFiles(java.util.List)",
         "org.zaproxy.zap.extension.api.WikiAPIGenerator#generateWikiFiles(java.util.List)",
         "org.zaproxy.zap.extension.ascan.ActiveScan#updatePluginRequestCounts()",
+        "org.zaproxy.zap.extension.ascan.CategoryTableModel#getQualtiy()",
```
Comment (thc202) [inline zap/zap.gradle.kts:270]: Not needed.

File: zap/zap.gradle.kts [hunk 5]
```diff
@@ -278,6 +279,7 @@ val japicmp by tasks.registering(JapicmpTask::class) {
         "org.zaproxy.zap.extension.history.PopupMenuPurgeSites#purge(org.parosproxy.paros.model.SiteMap,org.parosproxy.paros.model.SiteNode)",
         "org.zaproxy.zap.extension.pscan.ExtensionPassiveScan#addPassiveScanner(java.lang.String)",
         "org.zaproxy.zap.extension.pscan.PassiveScanThread#PassiveScanThread( org.zaproxy.zap.extension.pscan.PassiveScannerList, org.parosproxy.paros.extension.history.ExtensionHistory, org.zaproxy.zap.extension.alert.ExtensionAlert)",
+        "org.zaproxy.zap.extension.pscan.PolicyPassiveScanTableModel#getQuality()",
```
Comment (thc202) [inline zap/zap.gradle.kts:282]: Same here.

**Comment by kingthorin** [comment]
I think I got all those.

**Review by psiinon — APPROVED** [review]

**Comment by kingthorin** [comment]
Tweaked

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**